drivers/digit7seg/digit7seg.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2024 Orange
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_digit7seg
+ * @file
+ * @brief       Device driver for less than 5 digits of 7 segments without IC
+ * @author      Pierre Le Meur <pierre1.lemeur@orange.com>
+ */
+
+#include <assert.h>
+
+#define ENABLE_DEBUG        0
+#include "debug.h"
+
+#include "digit7seg.h"
+
+#define PIN_A               (dev->params.data_a)
+#define PIN_B               (dev->params.data_b)
+#define PIN_C               (dev->params.data_c)
+#define PIN_D               (dev->params.data_d)
+#define PIN_E               (dev->params.data_e)
+#define PIN_F               (dev->params.data_f)
+#define PIN_G               (dev->params.data_g)
+#define PIN_DP              (dev->params.data_dp)
+#define PIN_DIG1            (dev->params.dig1)
+#define PIN_DIG2            (dev->params.dig2)
+#define PIN_DIG3            (dev->params.dig3)
+#define PIN_DIG4            (dev->params.dig4)
+
+/** The 7 segments + decimal point + the number of digits */
+#define NB_PIN              (8+dev->params.digits)
+#define BYTE_BITS           (0x08)
+
+static void _set_pin_value(digit7seg_t *dev)
+{
+    const gpio_t pins[] =
+    {
+        PIN_A, PIN_B, PIN_C, PIN_D, PIN_E, PIN_F, PIN_G, PIN_DP
+    };
+
+    uint8_t mask = 0xFF;
+    uint8_t current_value = (uint8_t)(dev->value >> (dev->current_digit * 8) & mask);
+
+    for (int i = 0; i < BYTE_BITS; i++) {
+        if (current_value & (1 << i)) {
+            gpio_set(pins[i]);
+        }
+        else {
+            gpio_clear(pins[i]);
+        }
+    }
+}
+
+static void _shift_display(void *arg, int chan)
+{
+    (void)chan;
```

Comment:
why having the variable in this private function if it is unused?

Reply:
timer_init want a `timer_cb_t` that is : ` typedef void (*timer_cb_t)(void *arg, int channel);`